dataframe-jdbc/src/main/kotlin/org/jetbrains/kotlinx/dataframe/io/db/H2.kt

```diff
@@ -10,33 +10,67 @@ import kotlin.reflect.KType
 /**
  * Represents the H2 database type.
  *
- * This class provides methods to convert data from a ResultSet to the appropriate type for H2,
+ * This class provides methods to convert data from a ResultSet to the appropriate type for H2
  * and to generate the corresponding column schema.
  *
- * NOTE: All date and timestamp related types are converted to String to avoid java.sql.* types.
+ * NOTE: All date and timestamp-related types are converted to String to avoid java.sql.* types.
  */
-public object H2 : DbType("h2") {
+public class H2(public val dialect: DbType = MySql) : DbType("h2") {
```

Comment:
Actually, I might suspect the the dialect to be a subtype of H2. like `H2<MySl>()`. This could then later allow us to define functions on ony H2 databases with MySql dialect, for instance. If it's easier like this or you don't plan any extensions on it, then it's probably fine like this.